### PR TITLE
Google Workspace Admin Integration

### PR DESCRIPTION
**Pull request by anas-yousef**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: Not relevant

## Description
The Google Workspace Admin integration gives the user to options to view all mobile and chromeOS devices in the organization, and also perform remote actions on them.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: Packs/GoogleWorkspaceAdmin/Integrations/GoogleWorkspaceAdmin/GoogleWorkspaceAdmin_test.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+
+def create_test_client(mocker) -> Client:
+    """ This will create a mock client in order to use in the tests
+
+    Returns:
+        Client: A mock client instance
+    """
+    mocker.patch('GoogleWorkspaceAdmin.Client._init_credentials', return_value=None)
+    return Client(base_url=BASE_URL, verify=False, proxy=False, customer_id='id', service_account_json={})
```
Comment (JasBeilin) [inline Packs/GoogleWorkspaceAdmin/Integrations/GoogleWorkspaceAdmin/GoogleWorkspaceAdmin_test.py:31]:
```suggestion
    return Client(base_url='https://example.com/', verify=False, proxy=False, customer_id='id', service_account_json={})
```

**Review by JasBeilin — CHANGES_REQUESTED** [review]
1. Ask for image for the pack.
2. remove pipfile & pipfile.lock
3. add pack readme
4. change unitest according to the new code changes. -> make sure to check helper function seperatly, and make sure the CommandResult testing contains the call arguments and not the content of the commandResult itself wherever you have logic.

File: Packs/GoogleWorkspaceAdmin/Integrations/GoogleWorkspaceAdmin/GoogleWorkspaceAdmin.yml [hunk 2]
```diff
@@ -0,0 +1,616 @@
...
+      description: "Retrieves a paginated list that includes company-owned mobile devices."
+      arguments:
+        - auto: PREDEFINED
+          name: projection
+          description: "Whether to show all metadata fields, or only the basic metadata fields (e.g., deviceId, model, type, and status)"
+          required: false
+          isArray: false
+          predefined:
+            - BASIC
+            - FULL
```
[the remark this answers is not in the document]
Comment (DanilVil) [reply Packs/GoogleWorkspaceAdmin/Integrations/GoogleWorkspaceAdmin/GoogleWorkspaceAdmin.yml:62]: If it returns the same then it is a bug. I haven't tested it, @OmriItzhak did you?

File: Packs/GoogleWorkspaceAdmin/Integrations/GoogleWorkspaceAdmin/GoogleWorkspaceAdmin_description.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+In order to retrieve the Customer ID, do the following:
+1. In the Admin console, go to Menu => Account => Account settings => Profile.
+2. Under Customer ID, you will find your organization's unique ID.
+
+To execute the commands with the help of the service account, perform the following steps:
+1. You must be signed in as a super administrator for this task.
+2. Open your Google Admin console (at https://admin.google.com).
+3. Go to Admin roles.
+4. Click the role you want to assign (the role must have the appropriate privileges in order to execute the commands).
+5. The necessary privileges: Admin console privileges => Services => Mobile Device Management => Manage Devices and Settings.
```
[the remark this answers is not in the document]
Comment (anas-yousef) [reply Packs/GoogleWorkspaceAdmin/Integrations/GoogleWorkspaceAdmin/GoogleWorkspaceAdmin_description.md:13]: It does not work with ChromeOS only privileges, apparently the Mobile Device Management => Manage Devices and Settings privileges is enough for both mobile and ChromeOS devices.

**Pull request closed without merging**